application-analytics-ui/src/main/resources/Analytics/Code/Macros/MostViewedPages.xml

```diff
@@ -0,0 +1,799 @@
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.StringClass</classType>
+      </name>
+      <priority>
+        <disabled>0</disabled>
+        <name>priority</name>
+        <number>11</number>
+        <numberType>integer</numberType>
+        <prettyName>Priority</prettyName>
+        <size>10</size>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.NumberClass</classType>
+      </priority>
+      <supportsInlineMode>
+        <disabled>0</disabled>
+        <displayFormType>select</displayFormType>
+        <displayType>yesno</displayType>
+        <name>supportsInlineMode</name>
+        <number>5</number>
+        <prettyName>Supports inline mode</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </supportsInlineMode>
+      <visibility>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>visibility</name>
+        <number>6</number>
+        <prettyName>Macro visibility</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator>|</separator>
+        <separators>|</separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>Current User|Current Wiki|Global</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </visibility>
+    </class>
+    <property>
+      <async_cached>0</async_cached>
+    </property>
+    <property>
+      <async_context/>
+    </property>
+    <property>
+      <async_enabled>0</async_enabled>
+    </property>
+    <property>
+      <code>{{velocity}}
+#set ($discard = $xwiki.jsx.use('Analytics.Code.Macros.MostViewedPages'))
+#set ($object =  $xwiki.getDocument("Analytics.Code.Configuration").getObject('Analytics.Code.ConfigurationClass'))
+#set ($ip = $object.getProperty("requestAddress").value)
+#set ($idSite = $object.getProperty("idSite").value)
+#set ($authToken = $object.getProperty("authToken").value)
+#set ($dashbord =  $doc.getObject('Analytics.Code.AnalyticsDashboardClass'))
+#if ($dashbord)
+  #set ( $startTime =  $dashbord.getProperty("startDate"))
+  #set ( $endTime =  $dashbord.getProperty("endDate"))
+#else
+  #set ( $startTime = $xcontext.macro.params.startDate.toString() )
+  #set ( $endTime = $xcontext.macro.params.endDate.toString() )
+#end
+#set ($rows = $xcontext.macro.params.numberOfRows )
+#set ($index =$startTime.indexOf(' '))
+#if ($index != -1)
+  #set ($startTime = $startTime.substring(0, $index))
+#end
+#set ($index =$endTime.indexOf(' '))
+#if ($index != -1)
+  #set ($endTime = $endTime.substring(0, $index))
+#end
+
+{{html}}
+  &lt;div class="mostViewedPages" ip=$ip id=$idSite startDate=$startTime endDate=$endTime auth=$authToken&gt;
```

Comment:
[Data attributes](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) should be used for storing information. 
Also, I'd use more explicit name for each of them. For example, for `idSide`, don't shorten it to just id, since this already means something else. The same for `auth`, which without `token` could have a different, and not explicit, meaning.